File: services/backend/src/Apptivity.Domain/Entities/Report.cs
using Apptivity.Domain.Common;
using Apptivity.Domain.Enums;

namespace Apptivity.Domain.Entities;

public sealed class Report : BaseEntity
{
    public Guid ReporterId { get; set; }
    public Account Reporter { get; set; } = null!;
    public Guid TargetId { get; set; }
    public ReportTargetType TargetType { get; set; }
    public ReportReasonCategory ReasonCategory { get; set; }
    public required string Description { get; set; }
    public ReportStatus Status { get; set; } = ReportStatus.Pending;
}

File: apps/mobile/src/types/report.ts
export type ReportTargetType = 1 | 2;
export type ReportReasonCategory = 1 | 2 | 3 | 4 | 5 | 6;

export interface ReportRequest {
  targetId: string;
  targetType: ReportTargetType;
  reasonCategory: ReportReasonCategory;
  description?: string;
  evidenceImageUrl?: string;
}

export interface ReportReasonOption {
  label: string;
  value: ReportReasonCategory;
}

export interface ReportImageAsset {
  uri: string;
  fileName?: string | null;
  mimeType?: string | null;
}
